app/jobs/ants/sync_appointment_job.rb

```diff
@@ -24,18 +29,8 @@ def perform(ants_pre_demande_number:)
         appointment["management_url"].start_with?("#{protocol}://#{Domain::RDV_MAIRIE.host_name}")
       end
 
-      rdv = Rdv
-        .joins(:users)
-        .joins(motif: [:motif_category])
-        .merge(MotifCategory.requires_ants_predemande_number)
-        .where(users: { ants_pre_demande_number: [stripped_ants_pre_demande_number, ants_pre_demande_number] })
-        .where.not(status: Rdv::CANCELLED_STATUSES)
-        .where("starts_at >= ?", Time.zone.now)
-        .order(id: :desc) # choix arbitraire pour éviter un comportement aléatoire
-        .first
-
-      # on ne fait rien si les infos sont déjà identiques
-      return true if ants_appointments == [rdv&.serialize_for_ants_api]
+      # on ne fait rien si les infos sont déjà identiques (chacune des deux listes peuvent être vides ici)
+      return true if ants_appointments == [upcoming_rdv_serialized_to_ants_appointment]
 
       # on déclenche la suppression des appointments existants dans tous les cas, qu’il s’agisse d’une mise à jour ou d’une suppression
       # en effet l’API de l’ANTS ne permet pas de faire de mises à jour, on fait donc un delete puis un update
```

Comment:
Au passage, ce commentaire est incorrect, non ?

```suggestion
      # en effet l’API de l’ANTS ne permet pas de faire de mises à jour, on fait donc un delete puis un create
```